nucypher/cli/commands/multisig.py

```diff
@@ -0,0 +1,362 @@
+    """
+    Sign a proposed transaction before being sent to the MultiSig contract for execution
+    """
+    # Init
+    emitter = general_config.emitter
+    #_ensure_config_root(actor_options.config_root)
+    blockchain = blockchain_options.connect_blockchain(emitter, general_config.debug)
+    registry = blockchain_options.get_registry()
+
+    if not proposal:
+        raise click.MissingParameter("nucypher multisig sign requires the use of --proposal")
+
+    proposal = Proposal.from_file(proposal)
+
+    if not multisig_options.checksum_address:
+        multisig_options.checksum_address = select_client_account(emitter=emitter,
+                                                                  provider_uri=blockchain_options.provider_uri,
+                                                                  poa=blockchain_options.poa,
+                                                                  network=blockchain_options.network,
+                                                                  registry=registry,
+                                                                  show_balances=True)
+
+    name, version, address, abi = registry.search(contract_address=proposal.target_address)
+    # TODO: This assumes that we're always signing proxy retargetting. For the moment is true.
+    proxy_contract = blockchain.client.w3.eth.contract(abi=abi,
+                                                       address=address,
+                                                       version=version,
+                                                       ContractFactoryClass=blockchain._contract_factory)
+    paint_multisig_proposed_transaction(emitter, proposal, proxy_contract)
+
+    click.confirm("Proceed with signing?", abort=True)
+
+    executive = multisig_options.create_transactingless_executive(registry)  # FIXME: Since we use a signer, don't ask for PW
+    authorization = executive.authorize_proposal(proposal)
+    emitter.echo(f"\nSignature received from {authorization.recover_executive_address(proposal)}:\n")
+    emitter.echo(f"{authorization.serialize().hex()}\n", bold=True, color='green')
+
+
+@multisig.command()
+@group_general_config
+@group_blockchain_options
+@group_multisig_options
+@click.option('--proposal', help="Filepath to a JSON file containing a multisig transaction data",
+              type=EXISTING_READABLE_FILE)
+def execute(general_config, blockchain_options, multisig_options, proposal):
+    """
+    Collect authorizations from executives and execute transaction through MultiSig contract
+    """
+    # Init
+    emitter = general_config.emitter
+    #_ensure_config_root(actor_options.config_root)
+    blockchain = blockchain_options.connect_blockchain(emitter, general_config.debug)
+    registry = blockchain_options.get_registry()
+
+    if not proposal:
+        raise click.MissingParameter("nucypher multisig execute requires the use of --proposal")
```

Comment:
Same comment as the one made for the `sign` method.

Reply:
✔️